python/pyarrow/_parquet.pyx

```diff
@@ -2408,3 +2411,77 @@ cdef class ParquetWriter(_Weakrefable):
             return result
         raise RuntimeError(
             'file metadata is only available after writer close')
+
+    @property
+    def properties(self):
+        cdef const WriterProperties* props_ptr
+        cdef WriterPropertiesWrapper result
+
+        props_ptr = &self.writer.get().properties()
+        result = WriterPropertiesWrapper.__new__(WriterPropertiesWrapper)
+        result.init(props_ptr)
+        return result
+
+    @property
+    def arrow_properties(self):
+        cdef ArrowWriterPropertiesWrapper result
+
+        result = ArrowWriterPropertiesWrapper.__new__(ArrowWriterPropertiesWrapper)
+        result.init(self._arrow_properties)
+        return result
+
+
+cdef class WriterPropertiesWrapper(_Weakrefable):
```

Comment:
Why call it "Wrapper"? We should just call it `WriterProperties`. If that produces a name clash with the C++ class, then we should rename the latter in the Cython declarations to `CWriterProperties` as we do for many other such cases.